channeldb/db.go

```diff
@@ -1345,46 +1345,62 @@ func (c *ChannelStateDB) RestoreChannelShells(channelShells ...*ChannelShell) er
 
 // AddrsForNode consults the graph and channel database for all addresses known
 // to the passed node public key.
-func (d *DB) AddrsForNode(nodePub *btcec.PublicKey) ([]net.Addr,
-	error) {
+func (d *DB) AddrsForNode(nodePub *btcec.PublicKey) (bool, []net.Addr, error) {
+	var (
+		// addrs holds the collection of deduplicated addresses we know
+		// of for the node.
+		addrs = make(map[string]net.Addr)
+
+		// known keeps track of if any of the backing sources know of
+		// this node.
+		known bool
```

Comment:
Just double checking: Knowing about a node but the node having zero addresses is a valid state, right? So we can't just replace `known` with `len(addrs) > 0`?

Reply:
yeah `known` can be true while num addrs is zero. so not replaceable afaict